In [1]:
import pandas as pd
import glob

# Create the mega file

In [5]:
excel_files = glob.glob('./NY/*.xlsx') #/Users/trananhngan/Downloads/vortyx/NY

In [7]:
print(excel_files[0])
print(len(excel_files))

./NY/NY_960,001-980,000_advancesearch16059353506697285140a24.xlsx
25


In [8]:
all_df = []
for file in excel_files:
    df = pd.read_excel(file)
    all_df.append(df)

# Concaternate 
mega = pd.concat(all_df, ignore_index = True)

In [9]:
mega.head()

Company Name Company Type D-U-N-S@ Number  \
0  WALL TO WALL SERVICE & SUPPLIES INC      Private     13-014-0460   
1           BOARDWALK DISTRIBUTION LLC      Private     10-417-1881   
2              FULTON STREET OWNER LLC      Private     11-204-6567   
3             OCEAN PARTNERS OF NY LLC      Private     10-138-0573   
4              A.S.K. ELECTRICAL CORP.      Private     03-025-5976   

  Import/Export Fax No Global Duns No Global Name Immediate Parent Duns No  \
0           NaN    NaN            NaN         NaN                      NaN   
1           NaN    NaN            NaN         NaN                      NaN   
2           NaN    NaN            NaN         NaN                      NaN   
3           NaN    NaN            NaN         NaN                      NaN   
4           NaN    NaN            NaN         NaN                      NaN   

  Immediate Parent Name   Latitude  ... Primary NAICS Code  \
0                   NaN  40.677716  ...             423850   
1                   NaN  40.714713  ...             424990   
2                   NaN  40.714884  ...             813990   
3                   NaN  40.609264  ...             999990   
4                   NaN  40.875890  ...             221118   

                           Primary NAICS Description Primary SIC Code  \
0  Service Establishment Equipment and Supplies M...         50870300   
1  Other Miscellaneous Nondurable Goods Merchant ...         51990000   
2  Other Similar Organizations (except Business, ...         86410500   
3                                                NaN         99992222   
4                   Other Electric Power Generation          49110000   

                             Primary SIC Description     Sales  \
0  Cleaning and maintenance equipment and supplies\n  $569,130   
1                            Nondurable goods, nec\n  $420,695   
2                    Dwelling-related associations\n  $371,158   
3  Duns Support Record, nonclassifiable establish...  $339,299   
4                                Electric services\n  $329,263   

  Employee All Sites Employee this Site Sales (Year 1)  \
0                  4                  4       $130,000   
1                  4                  4        $43,329   
2                  7                  7        $35,206   
3                  6                  6        $94,459   
4                  7                  7        $51,594   

  Employees Total (Year 1) Facility Size  
0                        4             0  
1                        1             0  
2                        3             0  
3                        7             0  
4                        3             0  

[5 rows x 51 columns]

In [10]:
NY_mega = mega[['Company Name', 'Line of Business', 'Year of Founding', 'Phone No']]
NY_mega.head()

Company Name                      Line of Business  \
0  WALL TO WALL SERVICE & SUPPLIES INC  WHOL SERVICE ESTABLISHMENT EQUIPMENT   
1           BOARDWALK DISTRIBUTION LLC                 WHOL NONDURABLE GOODS   
2              FULTON STREET OWNER LLC              CIVIC/SOCIAL ASSOCIATION   
3             OCEAN PARTNERS OF NY LLC           NONCLASSIFIED ESTABLISHMENT   
4              A.S.K. ELECTRICAL CORP.                     ELECTRIC SERVICES   

   Year of Founding        Phone No  
0            1984.0  (718) 723-5497  
1            2017.0             NaN  
2            2012.0             NaN  
3            2019.0             NaN  
4            2019.0             NaN

# Exploratory data analysis

In [11]:
NY_mega.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 499529 entries, 0 to 499528
Data columns (total 4 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Company Name      499529 non-null  object 
 1   Line of Business  499528 non-null  object 
 2   Year of Founding  498905 non-null  float64
 3   Phone No          240795 non-null  object 
dtypes: float64(1), object(3)
memory usage: 15.2+ MB


In [16]:
NY_mega.shape

(499529, 4)

In [17]:
# Show the number of null values in each column
print( NY_mega.isnull().sum())

Company Name             0
Line of Business         1
Year of Founding       624
Phone No            258734
dtype: int64


In [21]:
# Show rows with null values
null_rows = NY_mega[NY_mega.isnull().any(axis=1)]
display(null_rows)

Company Name  \
1                   BOARDWALK DISTRIBUTION LLC   
2                      FULTON STREET OWNER LLC   
3                     OCEAN PARTNERS OF NY LLC   
4                      A.S.K. ELECTRICAL CORP.   
5       VICTORY WORLDWIDE TRANSPORTATION, INC.   
...                                        ...   
499502                RATHMORE PROPERTIES, LLC   
499503                         RED LIQUOR, LLC   
499505                             RISSON INC.   
499512                    SIGN MASTER USA INC.   
499527                    YOGURT FANCY NP INC.   

                         Line of Business  Year of Founding        Phone No  
1                   WHOL NONDURABLE GOODS            2017.0             NaN  
2                CIVIC/SOCIAL ASSOCIATION            2012.0             NaN  
3             NONCLASSIFIED ESTABLISHMENT            2019.0             NaN  
4                       ELECTRIC SERVICES            2019.0             NaN  
5             LOCAL TRUCKING-WITH STORAGE            2009.0             NaN  
...                                   ...               ...             ...  
499502   NONRESIDENTIAL BUILDING OPERATOR               0.0             NaN  
499503            RET ALCOHOLIC BEVERAGES               0.0             NaN  
499505            MEDICAL DOCTOR'S OFFICE               0.0             NaN  
499512  MFG SIGNS/ADVERTISING SPECIALTIES               NaN  (917) 295-4083  
499527                       EATING PLACE               0.0             NaN  

[259012 rows x 4 columns]

# Data wrangling: missing values

In [22]:
# Drop rows where Phone No has Null values
NY_mega = NY_mega.dropna(subset = ['Phone No'])
print(NY_mega.isnull().sum())

Company Name          0
Line of Business      0
Year of Founding    278
Phone No              0
dtype: int64


In [23]:
print(len(NY_mega))

240795


Total number of rows before cleaning = 499529

There are 258734 null values in phone number, none in Company Name

After removing missing values, we are left with 240795 rows

# Data wrangling: duplicates

In [28]:
# Find duplicates in the Company Name column 
# Find all duplicate values in the "Company Name" column
duplicate_values = NY_mega['Company Name'][NY_mega['Company Name'].duplicated(keep=False)]

# Filter the DataFrame to include only rows with duplicate values in the "Company Name" column
duplicate_rows = NY_mega[NY_mega['Company Name'].isin(duplicate_values)]

duplicate_rows_sorted = duplicate_rows.sort_values(by='Company Name')

display(duplicate_rows_sorted)

Company Name               Line of Business  \
398344  0 0 0 0 1 HOUR EMERGENCY LOCKSMITH                   RET HARDWARE   
125139  0 0 0 0 1 HOUR EMERGENCY LOCKSMITH                   RET HARDWARE   
398346           001 EMERGENCY A LOCKSMITH                   RET HARDWARE   
398345           001 EMERGENCY A LOCKSMITH                   RET HARDWARE   
128321            024 HOUR A DAY LOCKSMITH                   RET HARDWARE   
...                                    ...                            ...   
328762    ZION EVANGELICAL LUTHERAN CHURCH         RELIGIOUS ORGANIZATION   
435227                              ZONIES                   EATING PLACE   
252143                              ZONIES                   EATING PLACE   
19695                 ZUBE INTERIORS, INC.  REUPHOLSTERY/FURNITURE REPAIR   
40373                 ZUBE INTERIORS, INC.  REUPHOLSTERY/FURNITURE REPAIR   

        Year of Founding        Phone No  
398344               0.0  (718) 414-2132  
125139               0.0  (347) 296-8179  
398346               0.0  (718) 414-2146  
398345               0.0  (347) 296-8022  
128321               0.0  (347) 296-8168  
...                  ...             ...  
328762            2010.0  (518) 234-3381  
435227            2010.0  (585) 292-9664  
252143            2009.0  (716) 882-9664  
19695             1996.0  (845) 557-0618  
40373             1996.0  (845) 557-0618  

[6681 rows x 4 columns]

In [29]:
NY_mega = NY_mega.drop_duplicates(subset=['Company Name'], keep = 'first') 

In [30]:
NY_mega = NY_mega.drop_duplicates(subset=['Phone No'], keep = 'first') 

# Prepare the final mega file 

In [32]:
NY_mega.shape

(229842, 4)

In [33]:
display(NY_mega)

Company Name  \
0       WALL TO WALL SERVICE & SUPPLIES INC   
7                      FIRST CHOICE POS INC   
9              OHLA SYSTEMS & ELECTRIC, LLC   
12                                  YUTORAY   
13                    THE RENTAL SYSTEM INC   
...                                     ...   
499523                TOTAL IT SERVICES INC   
499524                     TSANG REALTY INC   
499525            US TRAVEL CONSULTANTS INC   
499526                     WARWICK INFO LLC   
499528           Z & A HARDWARE SUPPLY INC.   

                            Line of Business  Year of Founding        Phone No  
0       WHOL SERVICE ESTABLISHMENT EQUIPMENT            1984.0  (718) 723-5497  
7                CUSTOM COMPUTER PROGRAMMING            2015.0  (718) 522-0909  
9                       ENGINEERING SERVICES            2018.0  (718) 554-2320  
12               NONCLASSIFIED ESTABLISHMENT               0.0  (646) 543-4435  
13                 REAL ESTATE AGENT/MANAGER            2018.0  (845) 269-7539  
...                                      ...               ...             ...  
499523             COMPUTER RELATED SERVICES               0.0  (347) 987-0044  
499524           APARTMENT BUILDING OPERATOR               0.0  (718) 377-5891  
499525                         TRAVEL AGENCY               0.0  (516) 822-2715  
499526                          EATING PLACE               0.0  (845) 258-1000  
499528                    WHOL DURABLE GOODS               0.0  (917) 881-8131  

[229842 rows x 4 columns]

In [34]:
NY_mega.reset_index(drop = True , inplace = True)
display(NY_mega)

Company Name  \
0       WALL TO WALL SERVICE & SUPPLIES INC   
1                      FIRST CHOICE POS INC   
2              OHLA SYSTEMS & ELECTRIC, LLC   
3                                   YUTORAY   
4                     THE RENTAL SYSTEM INC   
...                                     ...   
229837                TOTAL IT SERVICES INC   
229838                     TSANG REALTY INC   
229839            US TRAVEL CONSULTANTS INC   
229840                     WARWICK INFO LLC   
229841           Z & A HARDWARE SUPPLY INC.   

                            Line of Business  Year of Founding        Phone No  
0       WHOL SERVICE ESTABLISHMENT EQUIPMENT            1984.0  (718) 723-5497  
1                CUSTOM COMPUTER PROGRAMMING            2015.0  (718) 522-0909  
2                       ENGINEERING SERVICES            2018.0  (718) 554-2320  
3                NONCLASSIFIED ESTABLISHMENT               0.0  (646) 543-4435  
4                  REAL ESTATE AGENT/MANAGER            2018.0  (845) 269-7539  
...                                      ...               ...             ...  
229837             COMPUTER RELATED SERVICES               0.0  (347) 987-0044  
229838           APARTMENT BUILDING OPERATOR               0.0  (718) 377-5891  
229839                         TRAVEL AGENCY               0.0  (516) 822-2715  
229840                          EATING PLACE               0.0  (845) 258-1000  
229841                    WHOL DURABLE GOODS               0.0  (917) 881-8131  

[229842 rows x 4 columns]

In [36]:
# Export the resulting dataframe into an excel file 
NY_mega.to_excel('NY_mega.xlsx' , index = True)